haraka plugin redis error on connecting to redis

### system info

Haraka | Haraka.js — Version: 2.8.28
 --- | :---
Node | v16.14.2
OS | Linux Debian 5.10.102.1-microsoft-standard-WSL2 #1 SMP Wed Mar 2 00:30:59 UTC 2022 x86_64 GNU/Linux
openssl | OpenSSL 1.1.1n  15 Mar 2022

### Expected behavior
No error on connecting to redis server
### Observed behavior
```
loaded TLD files:
  1=1495
  2=8570
  3=2448
loaded 9192 Public Suffixes
loglevel: INFO
log format: DEFAULT
[WARN] [-] [core] smtp.ini.nodes unset, using 1, see https://github.com/haraka/Haraka/wiki/Performance-Tuning
Starting up Haraka version 2.8.28
[INFO] [-] [core] Loading plugins
[INFO] [-] [core] Loading plugin: tls
[INFO] [-] [core] loading tls.ini
[INFO] [-] [core] Loading plugin: auth/flat_file
[INFO] [-] [core] Loading plugin: auth/auth_bridge
[INFO] [-] [core] Loading plugin: dkim_sign
[INFO] [-] [core] Loading plugin: redis
[INFO] [-] [core] Loading plugin: queue/smtp_forward
[NOTICE] [-] [queue/smtp_forward] outbound enabled, will default to disabled in Haraka v3 (see #1472)
[INFO] [-] [redis] connected to redis://192.168.0.100:6379
[CRIT] [-] [server] Plugin redis timed out on hook init_master - make sure it calls the callback
[INFO] [-] [server]  hook=init_master plugin=redis function=init_redis_shared params="" retval=DENYSOFT msg="plugin timeout"
[ERROR] [-] [server] init_master returned error: plugin timeout
[NOTICE] [-] [core] Shutting down
```
### Steps to reproduce
add redis to config/plugin
create redis.ini in config

I am having trouble to connect redis with haraka in order to run haraka-plugin-watch. Even though I put redis.ini in config folder it did not load new config for host or port

--- a/.eslintrc.yaml
+++ b/.eslintrc.yaml
@@ -2,15 +2,13 @@ env:
   node: true
   es6: true
   mocha: true
+  es2020: true
 
 plugins: [ haraka ]
 
 extends: [ eslint:recommended, plugin:haraka/recommended ]
 
 root: true
-
-parserOptions:
-  ecmaVersion: 2020
 
 globals:
   OK: true

--- a/Changes.md
+++ b/Changes.md
@@ -1,8 +1,9 @@
 
 
-### 2.0.1 - 2022-05-23
+### 2.0.2 - 2022-05-23
 
 - fix: rename p* methods -> * (required in redis v4)
+- fix: add `await ...connect()` as is now required, fixes #32
 - dep(redis): bump 4.0 -> 4.1
 
 

--- a/index.js
+++ b/index.js
@@ -194,12 +194,13 @@ exports.redis_subscribe_pattern = async function (pattern) {
     if (this.redis) return // already subscribed?
 
     this.redis = await redis.createClient(this.redisCfg.pubsub)
+    await this.redis.connect()
 
-    await this.redis.subscribe(pattern);
-    this.logdebug(this, `subscribed to ${pattern}`);
+    await this.redis.pSubscribe(pattern);
+    this.logdebug(this, `pSubscribed to ${pattern}`);
 }
 
-exports.redis_subscribe = async function (connection, event) {
+exports.redis_subscribe = async function (connection) {
 
     if (connection.notes.redis) {
         connection.logdebug(this, `redis already subscribed`);
@@ -211,12 +212,13 @@ exports.redis_subscribe = async function (connection, event) {
     }, 3 * 1000);
 
     connection.notes.redis = await redis.createClient(this.redisCfg.pubsub)
+    await connection.notes.redis.connect()
 
     clearTimeout(timer);
 
     const pattern = this.get_redis_sub_channel(connection)
-    connection.notes.redis.subscribe(pattern, event);
-    connection.logdebug(this, `subscribed to ${pattern}`);
+    connection.notes.redis.pSubscribe(pattern);
+    connection.logdebug(this, `pSubscribed to ${pattern}`);
 }
 
 exports.redis_unsubscribe = async function (connection) {
@@ -229,5 +231,5 @@ exports.redis_unsubscribe = async function (connection) {
     const pattern = this.get_redis_sub_channel(connection)
     await connection.notes.redis.unsubscribe(pattern);
     connection.logdebug(this, `unsubsubscribed from ${pattern}`);
-    connection.notes.redis.disconnect();
+    connection.notes.redis.quit();
 }

--- a/package.json
+++ b/package.json
@@ -1,6 +1,6 @@
 {
   "name": "haraka-plugin-redis",
-  "version": "2.0.1",
+  "version": "2.0.2",
   "description": "Redis plugin for Haraka & other plugins to inherit from",
   "main": "index.js",
   "directories": {
@@ -10,10 +10,10 @@
     "redis": "^4.1.0"
   },
   "devDependencies": {
-    "eslint": "^8.12.0",
+    "eslint": "8",
     "eslint-plugin-haraka": "*",
     "haraka-test-fixtures": "*",
-    "mocha": "^9.2.0"
+    "mocha": "9"
   },
   "scripts": {
     "lint": "npx eslint *.js test/*.js",

